core/src/test/java/org/infinispan/functional/FunctionalStorageTypeTest.java

```diff
@@ -0,0 +1,29 @@
+package org.infinispan.functional;
+
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.configuration.cache.StorageType;
+import org.testng.annotations.Test;
+
+@Test(groups = "functional", testName = "functional.FunctionalEncoderTest")
+public class FunctionalStorageTypeTest extends FunctionalMapTest {
+   StorageType storageType;
+
+   @Override
+   protected void configureCache(ConfigurationBuilder builder) {
+      builder.memory().storageType(storageType);
+      super.configureCache(builder);
+   }
+
+   public Object[] factory() {
+      return new Object[]{
+            new FunctionalStorageTypeTest().storageType(StorageType.OFF_HEAP),
+            new FunctionalStorageTypeTest().storageType(StorageType.BINARY),
+            new FunctionalStorageTypeTest().storageType(StorageType.OBJECT),
```

Comment:
`OBJECT` shouldn't be needed as `FunctionalMapTest` already runs this.

Reply:
ok !